api/v1alpha1/capiprovider_conditions.go

```diff
@@ -0,0 +1,10 @@
+package v1alpha1
+
+import (
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// GetConditions return CAPI Provider conditions.
+func (p *CAPIProvider) GetConditions() clusterv1.Conditions {
```

Comment:
Should we also have `SetConditions`?

Reply:
Please check CAPI Operator PR. The plan is to use the generic interface from upstream, and it needs help reviewing. For now this GetConditions is only required to satisfy the interface in logic https://github.com/rancher-sandbox/rancher-turtles/pull/293/files#diff-ccaf660a8ccddd6d42cb87d6455fcab1ca110bf8a65a9c3fad9af8c15a89eaf9R90 I’d prefer to keep changes minimal.

Reply:
which PR?